applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -183,22 +278,40 @@ static void handle_dfu_start(const struct config_event *event)
 		cur_offset = 0;
 	}
 
-	int err = flash_area_open(PM_MCUBOOT_SECONDARY_ID, &flash_area);
-	if (err) {
-		LOG_ERR("Cannot open flash area (%d)", err);
+	if (!flash_area) {
+		int err = flash_area_open(PM_MCUBOOT_SECONDARY_ID, &flash_area);
 
-		flash_area = NULL;
-	} else if (flash_area->fa_size < img_length) {
+		if (err) {
+			LOG_ERR("Cannot open flash area (%d)", err);
+
+			flash_area = NULL;
+			return;
+		}
+	}
+
+	if (flash_area->fa_size < img_length) {
 		LOG_WRN("Insufficient space for DFU (%zu < %" PRIu32 ")",
 			flash_area->fa_size, img_length);
 
 		flash_area_close(flash_area);
 		flash_area = NULL;
 	} else {
+		/* Clear trailer */
```

Comment:
I personally don't like this - this code is clearing some magic area on flash that. Imagine if this will change in the future.
Maybe we can ask imgmgr or mcuboot guys to add zephyr code to clear this part or we should leave it for the bootloader.

Reply:
Discussed offline. Since we make background DFU, we may make DFU possible only after flash is erased in background. In such case flash will be erased only in background.